modules/instance_template/variables.tf

```diff
@@ -365,6 +365,12 @@ variable "enable_confidential_vm" {
   description = "Whether to enable the Confidential VM configuration on the instance. Note that the instance image must support Confidential VMs. See https://cloud.google.com/compute/docs/images"
 }
 
+variable "confidential_instance_type" {
+  type        = string
+  default     = null
+  description = "Defines the confidential computing technology the instance uses. If this is set to \"SEV_SNP\", var.min_cpu_platform will be automatically set to \"AMD Milan\"."
+}
```

Comment:
might consider conditional value checks, if expect string to match...

Reply:
Pro: quicker (than waiting for the API) and clear user feedback
Con: required timely on-going maintenance to add future values

Reply:
for now, let's resolve as follows:

- No need to add condition check
- add a link to the description that indicates where the user can find the valid values. I think you found it from another comment. that describes SEV and SEV_SNP